### ISSUE TITLE
duplicateItems directive does not survive call to setChoices

### ISSUE THREAD

**Issue by ojensen5115** [jshjohnson/Choices#150]
Consider the following set of circumstances:

1. Choices constructed with `duplicateItems: false` in its initial config
2. Choices are populated via `setChoices` (e.g. as the result of a backend query)
3. An item is chosen
4. Choices are replaced via `setChoices` (e.g. as the result of a second backend query), and the new set of choices includes the same choice as was chosen before

Expected behavior: the item chosen in step 3 is not displayed as an option and/or cannot be chosen a second time.

Actual behavior: the item chosen in step 3 is displayed as an option, and can be chosen for a second time in spite of disallowing duplicateItems in the config.

Use case: I'm using Choices to have a user select recipients for a private message, where the recipient can be any registered username. As the user begins typing, the typed string is sent to the server to find all usernames with this prefix, and the list is returned and applied.

My code looks something like this:

```
        var choices = new window['Choices'](newElem, {
            'removeItemButton': true,
            'duplicateItems': false,
            'placeholderValue': 'Start typing a username...',
            'noChoicesText': '...',
            'itemSelectText': '',
            'searchFloor': 2,   // triggers search on 3 or more characters
            'shouldSort': false,
        });

        var lookupTimeout = null;

        goog.events.listen(newElem, 'search', function(event) {
            clearTimeout(lookupTimeout);
            lookupTimeout = setTimeout(serverLookup, 50);
        });

        goog.events.listen(newElem, 'choice', function(event) {
            // clear choices in the menu
            choices['setChoices']([], 'id', 'text', true);
        });

        var serverLookup = function() {
            utils.ajax.post('/api/userlookup', {'q': choices['currentValue']}, function(event) {
                var results = JSON.parse(event.target.getResponse());
                choices['setChoices'](results, 'id', 'text', true);
            });
        }
```

So if I  want to send a message to "SuperLongUsername" I type "sup" and can then select from a relatively short list of elements, before moving on to the next recipient. The list of recipients of a message should not include duplicates, but if I type "sup" a second time I can select "SuperLongUsername" a second time, in spite of the duplicateItems config.

A hacky workaround could scan through `choices.currentState.items` before the call to `setChoices` and remove any existing elements from the server response.

**Comment by ojensen5115** [comment]
A bit of digging suggests this is because the `duplicateItems` config option is only checked in `_canAddItem` if the input is a text input.

**Comment by jshjohnson** [comment]
I made a change in the last release to `_canAddItem` which _may_ have solved this issue. That said, it sounds like you are after a flag to use when you use `setChoices` that will clear existing items, right? 

That would be a really easy thing to implement if so. Let me know what you think

**Comment by ojensen5115** [comment]
I don't think that's what I'm after here. Essentially the "problem" case (as seen in v2.7.8) is as follows:

- initialize Choices on a multiselect `<select>` tag with `duplicateItems` set to false.
- select an item.
- use setChoices to replace the available options with a new list of options. Have this new list of options include the option that you already selected.
- Choices does not prevent you from selecting the same option again.

Issue #162 describes a non-toy use case in which this is apparent. From my digging earlier it seemed like the `duplicateItems` option doesn't do anything if choices is initialized on a `select` tag. I'll test to see if the issue is resolved in your latest release tomorrow.

**Comment by jshjohnson** [comment]
Ahh I see - yeah I can see the problem now. Let me investigate 👍

**Comment by jshjohnson** [comment]
This should now be resolved as of [version 2.8.2](https://github.com/jshjohnson/Choices/releases/tag/v2.8.2)! 

Let me know if you have any issues
